lib/ansible/utils/path.py

```diff
@@ -103,3 +104,15 @@ def basedir(source):
         dname = os.path.abspath(dname)
 
     return to_text(dname, errors='surrogate_or_strict')
+
+
+def cleanup_tmp_dir(path):
+    """Removes temporary directory"""
+    if os.path.isdir(path):
+        try:
+            shutil.rmtree(path)
+        except Exception as e:
+            # Importing here to avoid circular import
+            from ansible.utils.display import Display
+            display = Display()
+            display.display('Unable to remove temporary file {0}'.format(to_native(e)))
```

Comment:
but this all in a try/except/pass to avoid leaving other temp dirs if one fails all way through

Reply:
So if something in the `display` code raises an exception, I should just `pass` to avoid leaving behind a tmp dir? But won't the tmp dir already be removed at this point? Or an exception raised here would prevent as yet unremoved temp dirs from being removed?

Reply:
take N tempdirs into account and this failing in  x < N, so N-x tempdirs will be left 'unattempted'